rholang/src/test/scala/coop/rchain/rholang/interpreter/NormalizeTest.scala

```diff
@@ -348,6 +348,88 @@ class ProcMatcherSpec extends FlatSpec with Matchers {
       ))))
     result.knownFree should be(inputs.knownFree)
   }
+
+  "PInput" should "Handle a simple receive" in {
+    val listBindings = new ListName()
+    listBindings.add(new NameVar("x"))
+    listBindings.add(new NameVar("y"))
+    val listLinearBinds = new ListLinearBind()
+    listLinearBinds.add(new LinearBindImpl(listBindings, new NameQuote(new PNil())))
+    val linearSimple = new LinearSimple(listLinearBinds)
+    val receipt      = new ReceiptLinear(linearSimple)
+
+    val listSend = new ListProc()
+    listSend.add(new PEval(new NameVar("y")))
+    val body       = new PSend(new NameVar("x"), new SendSingle(), listSend)
+    val basicInput = new PInput(receipt, body)
+    val freeCount  = 2
+
+    val result = ProcNormalizeMatcher.normalizeMatch(basicInput, inputs)
+    result.par should be(
+      inputs.par.prepend(Receive(
+        List((List(ChanVar(FreeVar(0)), ChanVar(FreeVar(1))), Quote(Par()))),
+        Send(ChanVar(BoundVar(0)), List[Par](Eval(ChanVar(BoundVar(1)))), false),
+        false, // persistent
+        freeCount
+      )))
+    result.knownFree should be(inputs.knownFree)
+  }
+  "PInput" should "Handle a more complicated receive" in {
```

Comment:
If this is correct, can you add this in as a comment? It takes a bit of time to parse this otherwise.

`// for ( (x1, @y1) <- @Nil ; (x2, @y2) <- @1) { x1!(y2) | x2!(y1) }`

In retrospect, I should have gave Pars the lowest score so that Nil would come first - feels unintuitive that 1 comes before Nil. Do you think that is worth refactoring for?

Reply:
It does make sense to me that an Empty Par should come first.
This is minor, and you shouldn't fix it until you find yourself bored.